### PR TITLE
Added basic features for points

### PR DESCRIPTION
**Pull request by jonathanmli**

### CODE REVIEW

File: core/point/point.go [hunk 1]
```diff
@@ -12,6 +12,12 @@ type Point struct {
 	y int64
 }
 
+// Coord is a float point for representing points in space
+type Coord struct {
```
Comment (artasparks) [inline core/point/point.go]: How about we move this to core/point/coord.go? Would make it a bit clearer, I think.

File: core/point/point.go [hunk 2]
```diff
@@ -52,6 +58,20 @@ func (pt *Point) X() int64 { return pt.x }
 // Y returns the y-value.
 func (pt *Point) Y() int64 { return pt.y }
 
+// NewCd creates a new immutable Coord.
+func NewCd(x, y float64) *Coord {
```
Comment (artasparks) [inline core/point/point.go]: Just NewCoord I think

**Comment by artasparks** [comment]
Let me know when you move the Coord and we can merge the PR =)

**Comment by artasparks** [comment]
ping =)

**Comment by jonathanmli** [comment]
done! (I think)

**Comment by artasparks** [comment]
I'm not seeing the new file. Check your `git status` -- I suspect there's some untracked files. So, I think you might need to do a `git add -A` and then `git commit` or `git commit -m <msg>`

**Review by artasparks — APPROVED** [review]
Looks good to me!

**Pull request merged**